### PR TITLE
Added Fiber executor to allow large scale parallel hyperparameter opt…

### PR DESCRIPTION
**Pull request by calio**
…imization

# Code Pull Requests

Please provide the following:

- a clear explanation of what your code dose
- if applicable, a reference to an issue
- a reproducible test for your PR (code, model definition and data sample)

# Documentation Pull Requests

Note that the documentation HTML files are in `docs/` while the Markdown sources are in `mkdocs/docs`.

If you are proposing a modification to the documentation you should change only the Markdown files.

`api.md` is automatically generated from the docstrings in the code, so if you want to change something in that file, first modify `ludwig/api.py` docstring, then run `mkdocs/code_docs_autogen.py`, which will create `mkdocs/docs/api.md` .

### CODE REVIEW

File: ludwig/utils/hyperopt_utils.py [hunk 1]
```diff
@@ -23,6 +24,7 @@
 from abc import ABC, abstractmethod
 from typing import Any, Dict, List
 
+import fiber
```
Comment (w4nderlust) [inline ludwig/utils/hyperopt_utils.py:27]: Can yu move this import in the init of the FiberExecutor? This way if one does not need to use the fiber executor there will be no import error. I believe you refer to fiber directly only in the init to create the pool

**Comment by w4nderlust** [comment]
This looks great @calio thank you so much.
Could you please provide some instructions for me to test it both locally and remotely?

File: ludwig/utils/hyperopt_utils.py (hunk 1 above)
Comment (calio) [reply ludwig/utils/hyperopt_utils.py:27]: sure, will do.

**Comment by w4nderlust** [comment]
This works like a charm :)

**Pull request merged**